src/curry.ts

```diff
@@ -80,7 +80,7 @@ const memoize = <T>(
  * is given previously computed values will be checked
  * for expiration before being returned.
  */
-export const memo = <TFunc extends Function>(
```

Comment:
Why change from `Function` to `(...args: any) => any`?

Reply:
@rayepps ah yes, that was ESLint's complaining. I forgot to change that...

Reply:
@rayepps now that I've rechecked that though, I think that it's better that we keep this change to prevent class declarations (due to how we call func - `func(...args)`).